src/plugins/json/jsonplugin.cpp

```diff
@@ -326,8 +326,35 @@ JsonTemplateGroupFormat::JsonTemplateGroupFormat(QObject *parent)
 
 Tiled::TemplateGroup *JsonTemplateGroupFormat::read(const QString &fileName)
 {
-    Q_UNUSED(fileName);
-    return nullptr;
+    QFile file(fileName);
+
+    if (!file.open(QIODevice::ReadOnly | QIODevice::Text)) {
+        mError = tr("Could not open file for reading.");
+        return new Tiled::TemplateGroup();
```

Comment:
Yeah that's wrong, I used tileset JSON reader as reference, where the returned shared pointer is actually a nullptr.